### PR TITLE
Psr 764

### PR DESCRIPTION
**Pull request by narendrapaduchuri**

### CODE REVIEW

File: app/pages/nonsipp/shares/WhyDoesSchemeHoldSharesPage.scala [hunk 1]
```diff
@@ -45,5 +46,9 @@ case class WhyDoesSchemeHoldSharesPage(srn: Srn, index: Max5000) extends Questio
     }
 
   private def dependantPages(srn: Srn): List[Removable[_]] =
-    List(WhenDidSchemeAcquireSharesPage(srn, index))
+    List(
```
Comment (cyeniceri) [inline app/pages/nonsipp/shares/WhyDoesSchemeHoldSharesPage.scala:49]: Do you need to update WhyDoesSchemeHoldSharesPageSpec after this change ?

File: app/navigation/nonsipp/SharesNavigator.scala [hunk 2]
```diff
@@ -213,120 +213,203 @@ object SharesNavigator extends JourneyNavigator {
...
     userAnswers => {
 
+      case TypeOfSharesHeldPage(srn, index) =>
+        userAnswers.get(TypeOfSharesHeldPage(srn, index)) match {
+          case Some(SponsoringEmployer) =>
+            controllers.nonsipp.shares.routes.WhyDoesSchemeHoldSharesController.onPageLoad(srn, index, CheckMode)
+          case Some(Unquoted) =>
+            controllers.nonsipp.shares.routes.WhyDoesSchemeHoldSharesController.onPageLoad(srn, index, CheckMode)
+          case Some(ConnectedParty) =>
+            controllers.nonsipp.shares.routes.SharesCYAController.onPageLoad(srn, index, CheckMode)
```
Comment (cyeniceri) [inline app/navigation/nonsipp/SharesNavigator.scala:223]: For all these navigation changes in check routes, do we need to add corresponding unit tests in SharesNavigatorSpec ?

File: app/pages/nonsipp/shares/WhyDoesSchemeHoldSharesPage.scala (hunk 1 above)
Comment (narendrapaduchuri) [reply app/pages/nonsipp/shares/WhyDoesSchemeHoldSharesPage.scala:49]: Hi @cyeniceri I believe we are not updating the this WhyDoesSchemeHoldSharesPageSpec.

File: app/navigation/nonsipp/SharesNavigator.scala (hunk 2 above)
Comment (narendrapaduchuri) [reply app/navigation/nonsipp/SharesNavigator.scala:223]: Yes, we need to fix @cyeniceri. Let me check if I can fix in this or next one

File: app/pages/nonsipp/shares/WhyDoesSchemeHoldSharesPage.scala (hunk 1 above)
Comment (kberec) [reply app/pages/nonsipp/shares/WhyDoesSchemeHoldSharesPage.scala:49]: I think github shows this change with a mistake. I did this change (and updated WhyDoesSchemeHoldSharesPageSpec too) and that change has gone into main already.
@narendrapaduchuri could you please rebase on main? That might remove this change for the PR.

**Review by cyeniceri — APPROVED** [review]

**Pull request merged**